text/convert.py

```diff
@@ -34,3 +37,119 @@ def run(self):
             for seq_tag, entry in d.items():
                 assert isinstance(entry, str), f"expected str, got {entry!r} (type {type(entry).__name__})"
                 out.write(entry + "\n")
+
+
+class TextDictToStmJob(Job):
+    """
+    Similar as :class:`CorpusToStmJob`, but does not use the Bliss XML, but instead the text dict as input
+    (e.g. via :class:`CorpusToTextDictJob`).
+    Also see :class:`SearchWordsDummyTimesToCTMJob`.
+    """
+
+    def __init__(
+        self,
+        text_dict: Path,
+        *,
+        remove_non_speech_tokens: Sequence[str] = (),
+        remove_punctuation_tokens: Union[str, Sequence[str]] = (),
+        fix_whitespace: bool = True,
+        tag_mapping: Sequence[Tuple[Tuple[str, str, str], Dict[int, Path]]] = (),
+        seg_length_time: float = 1.0,
+    ):
+        """
+        :param text_dict: e.g. via :class:`CorpusToTextDictJob`
+        :param remove_non_speech_tokens: defines the list of non speech tokens to remove
+        :param remove_punctuation_tokens: defines list/string of punctuation tokens to remove
+        :param fix_whitespace: should white space be fixed.
+            !!!be aware that the corpus loading already fixes white space!!!
+        :param tag_mapping: 3-string tuple contains ("short name", "long name", "description") of each tag.
+            and the Dict[int, Path] is e.g. the out_single_segment_files of a FilterSegments*Jobs
+        :param seg_length_time: length of each segment in seconds.
+            should be consistent to :class:`SearchWordsDummyTimesToCTMJob`
+        """
+        self.set_vis_name("Extract STM from text-dict file")
+
+        self.text_dict = text_dict
+        self.remove_non_speech_tokens = remove_non_speech_tokens
+        self.remove_punctuation_tokens = remove_punctuation_tokens
+        self.fix_whitespace = fix_whitespace
+        self.tag_mapping = tag_mapping
+        self.seg_length_time = seg_length_time
+
+        self.out_stm_path = self.output_path("corpus.stm")
+
+    def tasks(self):
+        yield Task("run", mini_task=True)
+
+    def run(self):
+        # nan/inf should not be needed, but avoids errors at this point and will print an error below,
+        # that we don't expect an N-best list here.
+        c = eval(uopen(self.text_dict, "rt").read(), {"nan": float("nan"), "inf": float("inf")})
+        assert isinstance(c, dict)
```

Comment:
Similar comment as in the other PR, would this make sense here too?

Reply:
Not sure. Even the job name indicates that this is for the text-dict format.

Reply:
Dont have a strong opinion on this, was just thinking if at some point we wanted to extend if its easier one way around. But we can keep it as is.